detox/src/devices/Device.js

```diff
@@ -13,10 +13,10 @@ class Device {
   }
 
   async prepare() {
+    await this.deviceDriver.prepare();
+
     this._deviceId = await this.deviceDriver.acquireFreeDevice(this._deviceConfig.device || this._deviceConfig.name);
     this._bundleId = await this.deviceDriver.getBundleIdFromBinary(this._deviceConfig.binaryPath);
-
```

Comment:
@LeoNatan be advised. The only current usage of this is by SimulatorDriver. I believe it would be better for it, too.